bundles/admin/admin-announcements/view/AnnouncementsListHandler.js

```diff
@@ -0,0 +1,156 @@
+            } else {
+                this.updateState({
+                    announcements: data.data
+                });
+            }
+        }.bind(this));
+    }
+
+    saveAnnouncement (data) {
+        this.announcementsHelper.saveAnnouncement(data, function (err) {
+            if (err) {
+                Messaging.error(getMessage('messages.saveFailed'));
+            } else {
+                Messaging.success(getMessage('messages.saveSuccess'));
+                this.updateState({
+                    activeKey: []
+                });
+                this.fetchAdminAnnouncements();
+            }
+        }.bind(this));
+    }
+
+    // Update all the announcements f.ex. when saved. Set active key as empty so all panels get closed.
+    updateAnnouncement (data) {
+        this.announcementsHelper.updateAnnouncement(data, function (err, data) {
+            if (err) {
+                Messaging.error(getMessage('messages.updateFailed'));
+                return false;
+            } else {
+                Messaging.success(getMessage('messages.updateSuccess'));
+                removeFromLocalStorageArray(ANNOUNCEMENTS_LOCALSTORAGE, data.id);
+                this.updateState({
+                    activeKey: []
+                });
+                this.fetchAdminAnnouncements();
+            }
+        }.bind(this));
+    }
+
+    deleteAnnouncement (index, id) {
+        this.id = { id };
+        this.announcementsHelper.deleteAnnouncement(this.id, function (err) {
+            if (err) {
+                Messaging.error(getMessage('messages.deleteFailed'));
+            } else {
+                Messaging.success(getMessage('messages.deleteSuccess'));
+                const newList = [...this.state.announcements];
+                newList.splice(index, 1);
+                // Update accordion with announcements and keep all closed
+                this.updateState({
+                    announcements: newList,
+                    activeKey: [],
+                    updated: false
+                });
+                this.fetchAdminAnnouncements();
+            }
+        }.bind(this));
+    }
+    // Cancel creating a new announcement or editing one. Close all panels.
+    cancel (index, id) {
+        if (id !== undefined) {
+            this.updateState({
+                activeKey: []
+            });
+        } else {
+            const newList = [...this.state.announcements];
+            newList.splice(index, 1);
+            this.updateState({
+                announcements: newList,
+                activeKey: []
+            });
+        }
+    }
+
+    addForm () {
+        const newList = [...this.state.announcements];
+        newList.push({
+            title: this.newTitle
+        });
+        this.updateState({
+            announcements: newList,
+            title: this.newTitle
+        });
+    }
+
+    updateTitle (value) {
```

Comment:
Why is title updated on it's own? The first announcement to add will have the "newTitle" as initial value but the second one will have the first ones title as initial value?

Reply:
Actually you can probably remove this function and just use the antd forms onFinish() like you already do now.

Reply:
This was supposed to update the title as title is written but it doesn't seem work and also it's not necessary so I will just remove it.